### PR TITLE
[WIP] oc_image module implements image-import

### PR DESCRIPTION
**Pull request by ivanhorvath**

### CODE REVIEW

File: ansible/roles/lib_openshift_3.2/build/ansible/oc_image.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+    if not module.params['image_name']:
+        module.fail_json(msg='Please specify a name when state is absent|present.')
+
+    if state == 'present':
+
+        ########
+        # Create
+        ########
+        if not Utils.exists(api_rval['results'], module.params['image_name']):
```
Comment (ivanhorvath) [inline ansible/roles/lib_openshift_3.2/build/ansible/oc_image.py:50]: As discussed, I'm working on improving this, so it checks more than just the name.

Comment (kwoodson) [reply ansible/roles/lib_openshift_3.2/build/ansible/oc_image.py:50]: This should be fine for v1.

File: ansible/roles/lib_openshift_3.2/build/ansible/oc_image.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        module.exit_json(changed=False, results=api_rval, state="present")
+
+    module.exit_json(failed=True,
+                     changed=False,
+                     results='Unknown state passed. %s' % state,
+                     state="unknown")
+
+# pylint: disable=redefined-builtin, unused-wildcard-import, wildcard-import, locally-disabled
+# import module snippets.  This are required
+from ansible.module_utils.basic import *
```
Comment (kwoodson) [inline ansible/roles/lib_openshift_3.2/build/ansible/oc_image.py]: wrap these lines in if name section so we can test outside of ansible

File: ansible/roles/lib_openshift_3.2/build/src/oc_image.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        if results['returncode'] != 0 and '"%s" not found' % self.image_name in results['stderr']:
+            results['returncode'] = 0
+
+        return results
+
+    def create(self, url=None, name=None, tag=None):
+        '''Create an image '''
+
+        image = '{0}:{1}'.format(name, tag)
+        cmd = [image, '--from={0}/{1}'.format(url, image), '-n{0}'.format(self.namespace), '--confirm']
```
Comment (kwoodson) [inline ansible/roles/lib_openshift_3.2/build/src/oc_image.py]: I'd prefer it if this logic was moved down inside of the _import_image call.  That way I can call import_image from another module if needed.

_import_image would take ` url=None, name=None, tag=None`

**Review by kwoodson — COMMENTED** [review]
Consider moving logic from create into the import_image.

**Review by kwoodson — APPROVED** [review]

**Pull request merged**